packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts

```diff
@@ -268,6 +268,80 @@ describeNoCompat("GC attachment blob tombstone tests", (getTestObjectProvider) =
+        ],
+        async () => {
+            const { dataStore: mainDataStore, summarizer } = await createDataStoreAndSummarizer();
+
+            // Upload an attachment blob.
+            const blobContents = "Blob contents";
+            const blobHandle = await mainDataStore._runtime.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Reference and then unreference the blob so that it's unreferenced in the next summary.
+            mainDataStore._root.set("blob1", blobHandle);
+            mainDataStore._root.delete("blob1");
+
+            // Summarize so that the above attachment blobs are marked unreferenced.
+            await provider.ensureSynchronized();
+            const summary1 = await summarizeNow(summarizer);
+
+            // Wait for half sweep timeout and load a container. This container will upload a blob with the same content
+            // as above so that it is de-duped. This container should be able to use this blob until its session
+            // expires.
+            await delay(sweepTimeoutMs / 2);
+            const container2 = await loadContainer(summary1.summaryVersion);
+            const container2MainDataStore = await requestFluidObject<ITestDataObject>(container2, "default");
+            // Upload the blob and keep the handle around until the blob uploaded by first container is tombstoned.
+            const container2BlobHandle =
+                await container2MainDataStore._runtime.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+
+            // Wait for sweep timeout so that the blob uploaded by the first container is tombstoned.
+            await delay(sweepTimeoutMs + 10);
+
+            // Send an op to update the current reference timestamp that GC uses to make sweep ready objects.
+            mainDataStore._root.set("key", "value");
+            await provider.ensureSynchronized();
+
+            // Summarize so that the tombstoned blob is now part of the summary.
+            const summary2 = await summarizeNow(summarizer);
+
+            // Load a container from this summary and upload a blob with the same content as the tombstoned blob.
+            // This blob will get de-duped but it should be fine to use it because from this container's perspective
+            // it uploaded a brand new blob.
+            const container3 = await loadContainer(summary2.summaryVersion);
+            const container3MainDataStore = await requestFluidObject<ITestDataObject>(container3, "default");
+
+            const container3BlobHandle =
+                await container3MainDataStore._runtime.uploadBlob(stringToBuffer(blobContents, "utf-8"));
+            // Ideally, this should not reject but currently it will because of a bug with how blob de-dup interacts
+            // with GC.
+            await assertDoesNotRejectInverse(container3BlobHandle.get(), "Container3 should be able to use the blob");
```

Comment:
Why can't we just use Container2?  I think I'd like to see an assert about what happens with Container2 when you try to get the blob handle.

Reply:
The blob can be retrieved in container2 because it is not tombstoned there.